sklearn/datasets/_california_housing.py

```diff
@@ -181,6 +179,9 @@ def fetch_california_housing(data_home=None, download_if_missing=True,
                                               feature_names,
                                               target_names)
 
+    if return_X_y:
```

Comment:
Moved this to better handle the case where both `return_X_y` and `as_frame` are `True`.  If they are both true, this will return `X` and `y` as pandas objects.

Reply:
TY to @wconnell for this comment => [(https://github.com/scikit-learn/scikit-learn/pull/15486#issuecomment-549087417)]